### PR TITLE
refinement bug

### PR DESCRIPTION
**Pull request by lucacarniato**

### CODE REVIEW

**Comment by ahmad-el-sayed** [comment]
@lucacarniato feature name includes GRIDEDIT-566. Issue is GRIDEDIT-556. Please remember this when you format the message of the squashed commits.

**Review by ahmad-el-sayed — CHANGES_REQUESTED** [review]

File: libs/MeshKernel/include/MeshKernel/MeshRefinement.hpp [hunk 1]
```diff
@@ -142,22 +142,31 @@ namespace meshkernel
         /// @brief Computes the edge and face refinement mask from sample values (compute_jarefine_poly)
         void ComputeRefinementMasksFromSamples();
 
-        /// @brief Computes the number of edges that should be refined in a face (compute_jarefine_poly)
+        /// @brief Computes refinement masks (compute_jarefine_poly)
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply libs/MeshKernel/include/MeshKernel/MeshRefinement.hpp:145]: I agree with putting this info in a comment. i will create a task for this

File: libs/MeshKernel/src/MeshRefinement.cpp [hunk 2]
```diff
@@ -1254,16 +1282,16 @@ void MeshRefinement::FindBrotherEdges()
...
             const auto secondEdgeOtherNode = OtherNodeOfEdge(m_mesh->m_edges[secondEdgeIndex], n);
+            const auto center = ComputeMiddlePointAccountingForPoles(m_mesh->m_nodes[firstEdgeOtherNode], m_mesh->m_nodes[secondEdgeOtherNode], m_mesh->m_projection);
 
             // compute tolerance
-            const auto firstEdgeSquaredLength = ComputeSquaredDistance(m_mesh->m_nodes[firstEdgeOtherNode], m_mesh->m_nodes[n], m_mesh->m_projection);
-            const auto secondEdgeSquaredLength = ComputeSquaredDistance(m_mesh->m_nodes[secondEdgeOtherNode], m_mesh->m_nodes[n], m_mesh->m_projection);
-            const auto squaredTolerance = 0.0000001 * std::max(firstEdgeSquaredLength, secondEdgeSquaredLength);
+            const auto firstEdgeSquaredLength = ComputeDistance(m_mesh->m_nodes[firstEdgeOtherNode], m_mesh->m_nodes[n], m_mesh->m_projection);
+            const auto secondEdgeSquaredLength = ComputeDistance(m_mesh->m_nodes[secondEdgeOtherNode], m_mesh->m_nodes[n], m_mesh->m_projection);
+            const auto tolerance = 1e-4 * std::max(firstEdgeSquaredLength, secondEdgeSquaredLength);
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply libs/MeshKernel/src/MeshRefinement.cpp]: changed to minConnectionDistance

File: libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridCreateUniform.cpp [hunk 3]
```diff
@@ -33,43 +33,92 @@
...
+std::vector<std::vector<meshkernel::Point>> CurvilinearGridCreateUniform::computeSpherical(const MakeGridParameters& makeGridParameters)
+{
+    std::vector result = computeCartesian(makeGridParameters);
+    if (result.empty())
+    {
+        return result;
+    }
+    const auto numM = result[0].size();
+    const auto numN = result.size();
+    constexpr double latitudeCloseToPoles = 88.0;
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply libs/MeshKernel/src/CurvilinearGrid/CurvilinearGridCreateUniform.cpp:87]: I wrote some comments, for later reference.

File: libs/MeshKernel/src/MeshRefinement.cpp [hunk 4]
```diff
@@ -824,36 +792,75 @@ std::tuple<size_t, size_t, size_t> MeshRefinement::FindHangingNodes(size_t face)
...
 
     if (IsEqual(m_interpolant->GetFaceResult(face), constants::missing::doubleValue))
     {
         return;
     }
 
+    size_t numEdgesToBeRefined = 0;
+    std::ranges::fill(m_refineEdgeCache, 0);
+
     if (m_refinementType == RefinementType::RefinementLevels)
```
[the remark this answers is not in the document]
Comment (lucacarniato) [reply libs/MeshKernel/src/MeshRefinement.cpp:864]: i am default initializing the refinement type in meshrefinement.hpp 

RefinementType m_refinementType = RefinementType::WaveCourant; ///< The type of refinement to use

**Review by ahmad-el-sayed — APPROVED** [review]

**Pull request merged**